### ISSUE TITLE
Fuji: exposures to rerun with desispec tag >=0.51.3

### ISSUE THREAD

**Issue by julienguy** [desihub/desispec#1647]
There are 57 fiberflatexp files (fiber flat correction per exposure) (from 40 exposures from 12 nights) for which the bug fixed in PR #1642 (and applied in tag 0.51.3) has resulted in an error on the fitted humidity >0.5% leading to an error the flat field correction >3.5% for some fibers. We may want to rerun those before releasing fuji/guadalupe. Here is the list:
```
20210410 00084241 b1
20210412 00084529 b4
20210504 00087257 b4
20210505 00087381 b1
20210509 00087820 b4
20210512 00088156 b1
20210619 00095297 b1
20210619 00095297 b3
20210619 00095297 b7
20210619 00095298 b1
20210619 00095298 b7
20210619 00095299 b1
20210619 00095299 b7
20210619 00095299 b8
20210619 00095300 b1
20210619 00095300 b7
20210619 00095301 b1
20210619 00095301 b7
20210619 00095301 b8
20210619 00095302 b1
20210619 00095302 b6
20210619 00095302 b7
20210619 00095302 b8
20210619 00095303 b1
20210619 00095303 b6
20210619 00095303 b7
20210619 00095304 b1
20210619 00095304 b3
20210619 00095304 b6
20210619 00095304 b7
20210619 00095305 b6
20210619 00095306 b6
20210619 00095307 b6
20210619 00095309 b6
20210619 00095310 b3
20210619 00095310 b4
20210619 00095311 b6
20210619 00095312 b8
20210619 00095313 b6
20210619 00095314 b4
20210626 00096198 b6
20210626 00096201 b6
20210626 00096202 b6
20210626 00096203 b6
20210626 00096204 b6
20210627 00096356 b4
20210706 00097666 b2
20210706 00097667 b2
20210706 00097675 b4
20210706 00097681 b2
20210707 00097847 b2
20210707 00097849 b2
20210707 00097850 b2
20210707 00097851 b2
20210707 00097852 b2
20210707 00097853 b2
20210709 00098061 b2
```

**Comment by akremin** [comment]
What is the anticipated impact of these changes? 

If the impact is large enough, these blue cameras will impact the r and z cameras in standard star fitting, right?

Technically this could even go as far as `desi_select_calib_stars` selecting different stars on other petals, but hopefully that won't be the case.

**Comment by julienguy** [comment]
Yes we may want to rerun the std star fit and the calibrations of the 3 cameras from the same spectrograph. But I don't think we need to rerun `desi_select_calib_stars`; no need to reprocess the other spectrographs of the same exposure. An we can restart after the extractions.

**Comment by sbailey** [comment]
I think the final impact would be less than 3.5% because systematic errors in the fiberflatfielding are partially absorbed into flux calibration

**Comment by sbailey** [comment]
A few notes:
  * I'm running a test prod of a few tiles on 20210619 to test whether this has any pragmatic downstream impact, given the sky subtraction and flux calibration should absorb some of the fiberflatfielding residual differences
  * This is a large enough list of exposures that it would be a significant distraction to address.  We're pretty well scripted for resubmitting crashes, but not for surgery on removing and rerunning individual petals of individual exposures, so there is real human development time to address this, which comes at the expense of attention to vetting the rest of fuji/guadalupe
  * the first 6 entries are the only ones in fuji=svda itself; if we decide we really do need to rerun we could do just those, and leave the guadalupe ones as-is
  * if we do re-run, I think we should do entire exposures post-extraction and not try save compute cycles by re-processing individual petals which messes up pipeline job tracking, leaves ambiguities about calibstars selection, etc.
  * my bias remains towards not rerunning these in the spirit of the perfect being the enemy of the good, but we'll see if the test prod reveals anything concerning

**Comment by sbailey** [comment]
FYI test jobs have finished in /global/cfs/cdirs/desi/users/sjbailey/spectro/redux/f8 covering
```
20210619 00095297 b1,b3,b7    -> tileid 20770
20210619 00095302 b1,b6,b7,b8 -> tileid 20389
20210619 00095303 b1,b6,b7    -> tileid 20389
20210619 00095310 b4          -> tileid 20929
20210619 00095312 b8          -> tileid 1916
20210619 00095314 b4          -> tileid 1514
```
I have not yet evaluated their outputs, but am posting this now in case someone else can take a look

**Comment by sbailey** [comment]
These test cases include the worst case 20210619 00095310 b4 which had a 17.7% difference in humidity correction between the two pipeline versions.  Although there is a ~1 electron difference in the sframe files around the collimator dip feature due to the different corrections that aren't completely absorbed by sky subtraction, both versions look very clean in the sky-subtracted sky fibers:

![image](https://user-images.githubusercontent.com/218471/152451704-344f3fdb-207e-4636-9e8a-b94afd0a0ed5.png)

Upper: sky-subtracted sky fibers for guadalupe and test prod;
Lower: their difference with a very different zscale (if plotted with the same zscale, it looks uniformly gray, i.e. the systematic difference is much less than the noise).

In the 5000 spectra on 10 petals tested here only 4 changed from ZWARN=0 to ZWARN=4, all with originally 9<DELTACHI2<13 dropping to <9 in the new prod.

The only significant redshift differences were ZWARN != 0 targets.  For the ZWARN=0 targets the redshift difference was <<1 km/s .

![image](https://user-images.githubusercontent.com/218471/152450282-794e1d0e-f2f7-453b-a6ad-39a0ac51ba6d.png)

Bottom line: I don't think this is worth rerunning for Fuji/Guadalupe.

**Comment by julienguy** [comment]
I did a visual inspection of several spectra from tile 20389. I do see a difference in the spectra for the fibers where the fiber flat correction is bad in fuji+ : it is slightly better in f8 but still bad. For fibers with the fiber flat correction is good, it's as good in fuji as in f8. So I agree we could not rerun those exposures. Example (top: fuji , bottom: f8), see the negative spike at ~4300A due to an incorrect flat field affecting the sky subtraction.

Fuji+
![Figure_2_old](https://user-images.githubusercontent.com/5192160/152577519-9c68db23-f8b1-4110-a2ee-932e9764b774.png)

f8
![Figure_2_new](https://user-images.githubusercontent.com/5192160/152577524-1b487008-fc05-4d7a-a135-69a6fc93113a.png)
